pkg/util/exclude_kubernetes.go

```diff
@@ -98,21 +103,62 @@ func ExcludeKubernetesResource(fs afero.Afero, basePath string, excludedResource
 		}
 	}
 
-	return fmt.Errorf("unable to find resource %s in %s or its bases", excludedResource, basePath)
+	return nil, fmt.Errorf("unable to find resource %s in %s or its bases", excludedResource, basePath)
 }
 
-// UnExcludeKubernetesResource finds a deleted resource in a child of the basePath and includes it again
-func UnExcludeKubernetesResource(fs afero.Afero, basePath string, unExcludedResource string) error {
+// for the provided base and all subbases, check each strategic merge patch and json patch
+// if they match the resid provided, remove them
+func ExcludeKubernetesPatch(fs afero.Afero, basePath string, excludedResource resid.ResId) error {
+	kustomization, err := getKustomization(fs, basePath)
+	if err != nil {
+		return errors.Wrapf(err, "exclude patch %s", excludedResource.String())
+	}
 
-	kustomizeYaml, err := fs.ReadFile(filepath.Join(basePath, "kustomization.yaml"))
+	newJSONPatches := []patch.Json6902{}
+	for _, jsonPatch := range kustomization.PatchesJson6902 {
+		if excludedResource.Gvk().Equals(jsonPatch.Target.Gvk) {
+			if jsonPatch.Target.Name == excludedResource.Name() {
+				// don't add to new patch list
+				continue
+			}
+		}
+		newJSONPatches = append(newJSONPatches, jsonPatch)
+	}
+	kustomization.PatchesJson6902 = newJSONPatches
+
+	newMergePatches := []patch.StrategicMerge{}
+	for _, mergePatch := range kustomization.PatchesStrategicMerge {
+		matches, err := mergePatchMatches(fs, basePath, string(mergePatch), excludedResource)
+		if err != nil {
+			return errors.Wrapf(err, "check if patch matches excluded resource")
+		}
+
+		if !matches {
+			newMergePatches = append(newMergePatches, mergePatch)
+		}
+	}
+	kustomization.PatchesStrategicMerge = newMergePatches
+
+	for _, base := range kustomization.Bases {
+		err = ExcludeKubernetesPatch(fs, filepath.Join(basePath, base), excludedResource)
+		if err != nil {
+			return errors.Wrapf(err, "base %s of %s", base, basePath)
```

Comment:
i really believe that when you wrap you should be basically echoing the function name assuming that function is named properly. something like the following:

```
return errors.Wrapf(err, "exclude kubernetes patch %s from base %s", base, basePath)
```